src/js/components/pages/scene-editor-gui.jsx

```diff
@@ -0,0 +1,793 @@
+            valid: true
+        };
+
+        if (mediaObject.style.position === "absolute") {
+            try {
+                console.log("SceneEditorGUI: Loading Position");
+                placement.x = parseFloat(mediaObject.style.left)
+                placement.y = parseFloat(mediaObject.style.top)
+                placement.width = parseFloat(mediaObject.style.width)
+                placement.height = parseFloat(mediaObject.style.height)
+                placement.isRandom = false
+
+
+                //get rotation
+                if(mediaObject.style.transform != null) {
+                    var rotateString = mediaObject.style.transform.match(/\(([^)]+)deg\)/)[1] //regex to extract rotate transform only
+                    placement.rotation=parseInt(rotateString);
+                } else {
+                    placement.rotation = 0;
+                }
+
+                //check to test old content that only has partial position.
+                if (isNaN(placement.y) || isNaN(placement.x) || isNaN(placement.width) || isNaN(placement.height)) {
+                    placement.valid = false; //partial postion can't be converted without loss of auther intent.
+                } else {
+                    placement.valid = true;
+                }
+
+            } catch (error) {
+                console.log("SceneEditorGUI: Error Loading Position", error);
+                placement.valid=false;
+            }
+
+        } else {
+            console.log("SceneEditorGUI: Position is Random");
+            placement.isRandom = true;
+            placement.valid = true;
+        }
+
+
+        console.log("SceneEditorGUI: Placement Loaded", placement);
+        return placement;
+    },
+
+    getImageSize(url, callback) {
+        var img = new Image();
+        img.src = url;
+        img.onload = function () {
+            callback(this.width, this.height);
+        }
+    },
+
+    // post render events: 
+    componentDidUpdate() {
+        //check for resize and update if size changed
+
+        if (this.SceneLayoutArea != null) {
+            area = this.SceneLayoutArea;
+            if (area.offsetHeight != this.state.height || area.offsetWidth != this.state.width) {
+                this.setState({width: area.offsetWidth, height: area.offsetHeight})
+            }
+        }
+
+        //check save flag and schedule save if required
+        if (this.state.shouldSave) {
+            this.setSave();
+        }
+
+    },
+
+    //get the best placement of an image while preserving aspect ratio
+    getDefaultPlacement() {
+          console.log("SceneEditorGUI: Getting Default Placement")
+          placement = this.state.placement;
+          if (this.state.mediaObject.type === "image") {
+            var self = this
+            this.getImageSize(self.state.mediaObject.url, function(imgWidth, imgHeight) {
+    
+                scaleX = imgWidth/self.state.width;
+                scaleY = imgHeight/self.state.height;
```

Comment:
Values are unused and the math seems switched 2 lines down - needs to be removed and declared

Reply:
Yep top 2 lines can be removed